boilerplate/fullstack/remix/app/lib/superTokensHelpers.ts

```diff
@@ -0,0 +1,121 @@
+import { PreParsedRequest, CollectingResponse } from "supertokens-node/framework/custom/index.js";
+import { HTTPMethod } from "supertokens-node/types";
+import Session, { SessionContainer, VerifySessionOptions } from "supertokens-node/lib/build/recipe/session/index.js";
+import SessionRecipe from "supertokens-node/lib/build/recipe/session/recipe.js";
+import { availableTokenTransferMethods } from "supertokens-node/lib/build/recipe/session/constants.js";
+import { getToken } from "supertokens-node/lib/build/recipe/session/cookieAndHeaders.js";
+import { parseJWTWithoutSignatureVerification } from "supertokens-node/lib/build/recipe/session/jwt.js";
+
+export function getCookieFromRequest(request: Request) {
+    const cookies: Record<string, string> = {};
+    const cookieHeader = request.headers.get("Cookie");
+    if (cookieHeader) {
+        const cookieStrings = cookieHeader.split(";");
+        for (const cookieString of cookieStrings) {
+            const [name, value] = cookieString.trim().split("=");
+            cookies[name] = decodeURIComponent(value);
+        }
+    }
+    return cookies;
+}
+
+export function getQueryFromRequest(request: Request) {
+    const query: Record<string, string> = {};
+    const url = new URL(request.url);
+    const searchParams = url.searchParams;
+    searchParams.forEach((value, key) => {
+        query[key] = value;
+    });
+    return query;
+}
+export function createPreParsedRequest(request: Request): PreParsedRequest {
+    return new PreParsedRequest({
+        cookies: getCookieFromRequest(request),
+        url: request.url as string,
+        method: request.method as HTTPMethod,
+        query: getQueryFromRequest(request),
+        headers: request.headers,
+        getFormBody: async () => {
+            return await request.formData();
+        },
+        getJSONBody: async () => {
+            return await request.json();
+        },
+    });
+}
+
+export async function getSessionDetails(
+    request: Request,
+    options?: VerifySessionOptions,
+    userContext?: Record<string, unknown>
+): Promise<{
+    session: SessionContainer | undefined;
+    hasToken: boolean;
+    hasInvalidClaims: boolean;
+    nextResponse?: Response;
```

Comment:
nextResponse is not a good name since this is not a nextjs app

Reply:
✔️